pkg/gds/gds_test.go

```diff
@@ -105,6 +106,160 @@ func (s *gdsTestSuite) TestRegister() {
+	request.Id = ""
+	request.CommonName = "invalid.name"
+	_, err = client.Lookup(ctx, request)
+	require.Error(err)
+}
+
+// TestSearch tests that the Search RPC returns the correct search results.
+func (s *gdsTestSuite) TestSearch() {
+	s.LoadFullFixtures()
+	require := s.Require()
+	ctx := context.Background()
+
+	// Start the gRPC client
+	require.NoError(s.grpc.Connect())
+	defer s.grpc.Close()
+	client := api.NewTRISADirectoryClient(s.grpc.Conn)
+
+	// No search criteria - should not return anything
+	request := &api.SearchRequest{}
+	reply, err := client.Search(ctx, request)
+	require.NoError(err)
+	require.Empty(reply.Error)
+	require.Len(reply.Results, 0)
+
+	// Search by name
+	request.Name = []string{"CharlieBank"}
+	reply, err = client.Search(ctx, request)
+	require.NoError(err)
+	require.Empty(reply.Error)
+	require.Len(reply.Results, 1)
+	id := "d9da630e-41aa-11ec-9d29-acde48001122"
+	vasp := s.fixtures[vasps][id].(*pb.VASP)
+	require.Equal(id, reply.Results[0].Id)
+	require.Equal(vasp.RegisteredDirectory, reply.Results[0].RegisteredDirectory)
+	require.Equal(vasp.CommonName, reply.Results[0].CommonName)
+	require.Equal(vasp.TrisaEndpoint, reply.Results[0].Endpoint)
+
+	// Multiple results
+	request.Name = []string{"CharlieBank", "Delta Assets"}
+	reply, err = client.Search(ctx, request)
+	require.NoError(err)
+	require.Empty(reply.Error)
+	require.Len(reply.Results, 2)
+
+	// Search by website
+	request = &api.SearchRequest{
+		Website: []string{"https://trisa.charliebank.io"},
+	}
+	reply, err = client.Search(ctx, request)
+	require.NoError(err)
+	require.Empty(reply.Error)
+	require.Len(reply.Results, 1)
+
+	// Filter by country
+	request = &api.SearchRequest{
+		Name:    []string{"CharlieBank"},
+		Country: []string{vasp.Entity.CountryOfRegistration},
+	}
+	reply, err = client.Search(ctx, request)
+	require.NoError(err)
+	require.Empty(reply.Error)
+	require.Len(reply.Results, 1)
+
+	// Filter by country - no results
+	request = &api.SearchRequest{
+		Name:    []string{"CharlieBank"},
+		Country: []string{"US"},
+	}
+	reply, err = client.Search(ctx, request)
+	require.NoError(err)
+	require.Empty(reply.Error)
+	require.Len(reply.Results, 0)
+
+	// Filter by category
+	request = &api.SearchRequest{
+		Name:             []string{"CharlieBank"},
+		BusinessCategory: []pb.BusinessCategory{vasp.BusinessCategory},
+	}
+	reply, err = client.Search(ctx, request)
+	require.NoError(err)
+	require.Empty(reply.Error)
+	require.Len(reply.Results, 1)
+
+	// Filter by business category - no results
+	request = &api.SearchRequest{
+		Name:             []string{"CharlieBank"},
+		BusinessCategory: []pb.BusinessCategory{pb.BusinessCategory_GOVERNMENT_ENTITY},
+	}
+	reply, err = client.Search(ctx, request)
+	require.NoError(err)
+	require.Empty(reply.Error)
+	require.Len(reply.Results, 0)
+
+	// Filter by VASP category
+	request = &api.SearchRequest{
+		Name:         []string{"CharlieBank"},
+		VaspCategory: []string{"Miner"},
```

Comment:
These VASP Category fields (e.g. "Miner", "Project", etc)  are also randomly selected by `fakerize.py`, so we might need to address these in sc-2282 too.